content/posts/why-meson/index.md

```diff
@@ -0,0 +1,250 @@
+- Many complex low-level pure C projects recently switched to Meson:
+[Mesa](https://mesa3d.org/),
+[Wayland](https://github.com/wayland-project/wayland),
+[PipeWire](https://github.com/PipeWire/pipewire),
+[QEMU](https://www.qemu.org/), and [many
+others](https://mesonbuild.com/Users.html). We are not alone in this!
+
+# Examples of using meson
+
+## Development process
+
+As a developer when you download Rizin, you can install it for your user in
+`~/.local`, so you don't need root access to install files. You can do this
+with `meson --prefix=~/.local build; ninja -C build install`. After that, you
+can change the source code however you need and then run `ninja` again with
+`ninja -C build`. Only the changed files are re-built.
+
+Moreover, running `ninja` by default builds files with explicit RPATHs, which
+means that the executables and libraries contain direct references to the
+paths of dependent libraries they are linked against so the loader can then
+always find them without having to specify `LD_LIBRARY_PATH` or similar. For
+this reason, most of the times you will not need to re-install the Rizin
+files, but while developing you can just run rizin from
+`./build/binrz/rizin/rizin`.
+
+RPATH are not, of course, always good. Indeed they are usually removed during
+the installation process. However, when you install Rizin in a place that is
+not `/usr`, we have chosen to keep RPATHs to make the installation process as
+simple as possible, without requiring users to mess with their environment to
+make sure the binaries can find the proper libraries. Packagers, who usually
+use `/usr` as prefix, should not be affected by this decision, but they can
+anyway disable it by specifying `-Dlocal=false` when running meson.
+
+## Reviewing a PR and testing changes
+
+When testing a PR with a fix or comparing multiple changes, you need to have
+access to multiple versions of Rizin. Doing this with ACR/Make is of course
+possible, but it usually involves installing everything in separated
+directories and making sure your environment variables (e.g. PATH,
+LD_LIBRARY_PATH, etc.) are correctly set. With meson you can build one
+version (e.g. from `dev` branch) with `meson --prefix=~/.local build-dev;
+ninja -C build-dev`, then switch branch with `git checkout my-other-branch`
+and build Rizin again with `meson --prefix=~/.local build-pr; ninja -C
+build-pr`. Due to the RPATH used by default, as mentioned above, each build
+directory can be used without installation to actually run the Rizin tools.
+At that point you can quickly compare the results of
+`./build-dev/binrz/rizin/rizin` and `./build-pr/binrz/rizin/rizin`.
+
+# Conclusion
+
+Of course it's not all perfect with meson either. Right now the meson build
+system is missing some features that were only available with ACR/Make.
+
+To uninstall Rizin you have to run `ninja -C build uninstall` from the same
+build directory you used to run the `install` step, otherwise it will not
```

Comment:
```suggestion
build directory you used to run the `install` step, otherwise, it will not
```

Reply:
I'm no native english speaker so...
I did not apply this change, I have some doubts about it. @ITAYC0HEN  ? Feel free to directly apply if you think it makes more sense.